cfgov/v1/models/base.py

```diff
@@ -212,12 +212,26 @@ def specific_categories_query(block, parent_slug):
         return {search_type: queryset for search_type, queryset in
                 related.items() if queryset}
 
+    def get_appropriate_page_version(self, request):
+        # If we're on the production site, make sure the version of the page
+        # displayed is the latest version that has `live` set to True for
+        # the live site or `shared` set to True for the staging site.
+        revisions = self.revisions.all().order_by('-created_at')
+        for revision in revisions:
+            page_version = json.loads(revision.content_json)
+            if not request.is_staging:
+                if page_version['live']:
+                    return revision.as_page_object()
+            else:
+                if page_version['shared']:
+                    return revision.as_page_object()
+
     def get_breadcrumbs(self, request):
         ancestors = self.get_ancestors()
         home_page_children = request.site.root_page.get_children()
         for i, ancestor in enumerate(ancestors):
             if ancestor in home_page_children:
-                return [util.get_appropriate_page_version(request, ancestor) for ancestor in ancestors[i+1:]]
+                return [ancestor.get_appropriate_page_version(request) for ancestor in ancestors[i+1:]]
```

Comment:
Seems like we need to call 'specific' here, since get_ancestors can return Page objects (which don't have the method). I included that fix on #2518